src/configuration.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::{anyhow, Result};
+use crate::cli::CliOptions;
+use dirs::config_dir;
+use merge::Merge;
+use std::{fs::{self, create_dir, File}, path::Path};
+use std::io::{BufReader, Read};
+
+pub fn read_cfg_file<P: AsRef<Path>>(path: &P) -> Result<CliOptions> {
+    let file = File::open(&path)?;
+    let mut buf_reader = BufReader::new(file);
+    let mut contents = String::new();
+    buf_reader.read_to_string(&mut contents)?;
+    Ok(toml::from_str(contents.as_str()).unwrap())
+}
+
+pub fn load_cfg<P: AsRef<Path>>(path: Option<&P>) -> Result<CliOptions> {
+    match path {
+        Some(config_path) => read_cfg_file(config_path),
+        None => {
+            let mut default_path = config_dir().unwrap();
+            default_path.push("/onefetch/config.toml");
+            if default_path.exists() {
+                read_cfg_file(&default_path)
+            } else {
+                create_dir(&default_path)?;
+                write_default_cfg(&default_path);
+                Err(anyhow!("Configuration file at {:?} does not exist!", default_path))
+            }
+        }
+    }
+}
```

Comment:
IMO this can just take a `P` instead of an `Option<P>`. Any unwrapping can happen before this is called, which can be simplified with:

```rust
path.unwrap_or_else(|| default_config_dir())
```

We may also want to replace some of these `unwrap`s and `expect`s with non-panicking `Result::Err` returns, instead.